File: WebApi/Models/Player.cs
using System;

namespace WebApi.Models
{
    public class Player
    {
        public double Money { get; set; }
        public double Income { get; set; }
        public double IncomePrice { get; set; }
        public int IncomeIncreases { get; set; }
        public double HealthPrice { get; set; }
        public Castle Castle { get; set; }
        public string Team { get; set; }
        public int Side { get; set; }

        public Player()
        {
            this.Money = 0;
            this.Income = 0.05;
            this.IncomeIncreases = 0;
            this.IncomePrice = 4;
            this.HealthPrice = 5;
        }
        public Player(string team, int side) : this()
        {
            this.Side = side;
            this.Team = team;
            this.Castle = new Castle(team, side);
        }

        public void GetMoney()
        {
            this.Money += this.Income;
        }
        public bool SpendMoney(double price)
        {
            if (this.Money < price)
                return false;
            this.Money -= price;
            return true;
        }

        public void AddIncome()
        {
            this.Income += 0.04;
            this.IncomeIncreases++;
            this.IncomePrice = Math.Pow(4.0, (this.IncomeIncreases + 5.0) / 5.0);
        }
        public void AddHealth()
        {
            this.Castle.Health += (int)(this.Castle.MaxHealth * 0.25);
            this.Castle.MaxHealth += (int)(this.Castle.MaxHealth * 0.25);
            this.HealthPrice += 2.5;
        }
    }
}

File: Angular/src/app/player.ts
import { Castle } from './game/castle';

export class Player {
    money: number;
    income: number;
    incomePrice: number;
    healthPrice: number;

    castle: Castle;
    side: number;

    constructor(money: number, income: number, incomePrice: number, healthPrice: number, side: number, castle: Castle) {
        this.money = money;
        this.income = income;
        this.incomePrice = incomePrice;
        this.healthPrice = healthPrice;
        this.side = side;
        this.castle = castle;
    }

    public updatePlayer(data, id: number) {
        if (typeof data[id] == 'undefined')
            return;
        
        this.money = data[id].money;
        this.income = data[id].income;
        this.incomePrice = data[id].incomePrice;
        this.healthPrice = data[id].healthPrice;
        this.side = data[id].side;
        this.castle = Castle.parseCastle(data[id].castle);
    }
    public static parsePlayer(data, id: number): Player {
        return new Player(data[id].money, data[id].income, data[id].incomePrice, data[id].healthPrice, data[id].side, Castle.parseCastle(data[id].castle));
    }
}